CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,133 @@
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+#include <QDebug>
+#include <QTimer>
+
+#include "../../InfrastructureLayer/Settings/I_Settings.h"
+#include "InternetConnectionService.h"
+
+namespace
+{
+    quint32 SLEEP_TIME_MILLISECONDS = 2000;
+}
+
+InternetConnectionService::InternetConnectionService(
+    QString exchangeName
+    ,  QString queueName
+    ,  QString ipAddress
+    ,  quint16 port)
+    : exchangeName_(exchangeName)
+    , queueName_(queueName)
+    , ipAddress_(ipAddress)
+    , port_(port)
+    , connectionRetryTimer_(new QTimer(this))
+{
+    QObject::connect(this, SIGNAL(setupChannelSignal()), this, SLOT(setupChannel()));
+
+    connectionRetryTimer_.setSingleShot(true);
+    connect(&connectionRetryTimer_, SIGNAL(timeout()), this, SLOT(setupChannel()));
+    connectToDataSource();
+}
+
+InternetConnectionService::~InternetConnectionService()
+{
+}
+
+
+void InternetConnectionService::setupChannel()
+{
+    qWarning() << "InternetConnectionService: Attempting to connect to " << ipAddress_ << port_;
+
+    try
+    {
+        channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), port_);
+        channel_->DeclareExchange(exchangeName_.toStdString(), AmqpClient::Channel::EXCHANGE_TYPE_FANOUT);
+        channel_->DeclareQueue(queueName_.toStdString());
+        channel_->BindQueue(queueName_.toStdString(), exchangeName_.toStdString());
+    }
+    catch (AmqpClient::ChannelException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << "Connection failed, retrying in" << SLEEP_TIME_MILLISECONDS << "ms";
+            connectionRetryTimer_.start(SLEEP_TIME_MILLISECONDS);
+            return;
+        }
+        else
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::ChannelErrorException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << "Connection failed, retrying in" << SLEEP_TIME_MILLISECONDS << "ms";
+            connectionRetryTimer_.start(SLEEP_TIME_MILLISECONDS);
+            return;
+        }
+        else
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel Error Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::AmqpResponseLibraryException&)
+    {
+        if (channel_ == NULL)
```

Comment:
I would remove all these checks for `channel_ == NULL`, first of here if channel is not null but there is an exception, the function just exits silently. Erroring for any exception should be the default.